frontend/src/components/maps/leaflet/LayerPopup/hooks/useLayerQuery.tsx

```diff
@@ -13,6 +13,46 @@ import { toast } from 'react-toastify';
 import { layerData } from 'constants/toasts';
 import * as rax from 'retry-axios';
 
+const MAX_RETRIES = 2;
+const wfsAxios = () => {
+  const instance = axios.create({ timeout: 5000, withCredentials: true });
+  instance.defaults.raxConfig = {
+    retry: MAX_RETRIES,
+    instance: instance,
+    shouldRetry: (error: AxiosError) => {
+      const cfg = rax.getConfig(error);
+      if (cfg?.currentRetryAttempt === MAX_RETRIES) {
+        toast.dismiss(layerData.LAYER_DATA_LOADING_ID);
+        layerData.LAYER_DATA_ERROR();
+      }
+      return rax.shouldRetryRequest(error);
+    },
+  };
+  rax.attach(instance);
+
+  instance.interceptors.request.use(config => {
+    layerData.LAYER_DATA_LOADING();
+    return config;
+  });
+
+  instance.interceptors.response.use(
+    response => {
+      return response;
+    },
+    error => {
+      if (axios.isCancel(error)) {
+        return Promise.resolve(error.message);
+      }
+      toast.dismiss(layerData.LAYER_DATA_LOADING_ID);
+      layerData.LAYER_DATA_ERROR();
+
+      // Error is handled and returning empty object.
+      return Promise.resolve({});
```

Comment:
@LawrenceLau2020 This is the line I'm thinking should be updated to handle the null value.  It should return an object that has the necessary properties.

Reply:
I'm still getting that error after trying this:

      // Error is handled and returning empty object.
      return Promise.resolve({
        features: [],
      });

Reply:
It look like the showLocationDetails in the Map.tsx is making a call to the userLayerQuery.tsx on line 143 shown below as I can't create a comment in that area of the code:

const findOneWhereContains = useCallback(
    async (latlng: LatLng): Promise<FeatureCollection> => {
      const data: FeatureCollection = (
        await wfsAxios().get(
          `${baseUrl}&cql_filter=CONTAINS(${geometryName},SRID=4326;POINT ( ${latlng.lng} ${latlng.lat}))`,
        )
      )?.data;
      return data;
    },
    [baseUrl, geometryName],
  );

There seems to be a null check for the data, but if there is no data it won't return anything, so maybe we need to somehow return an empty "FeatureCollecction" object from here?

Reply:
That makes sense to me.  
You can use the following syntax.
`?.data ?? { features: [] };`

Reply:
Sounds good, I'll add it to both the parcel request "findOneWhereContains" and the municipality request "findByAdministrative" as that function is almost identical.